lib/jwt/jwk/ec.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+module JWT
+  module JWK
+    class EC
+      extend Forwardable
+      def_delegators :@keypair, :private?, :public_key
+
+      attr_reader :keypair
+      attr_reader :kid
+
+      KTY    = 'EC'.freeze
+      BINARY = 2
+
+      def initialize(keypair, kid = nil)
+        raise ArgumentError, 'keypair must be of type OpenSSL::PKey::EC' unless keypair.is_a?(OpenSSL::PKey::EC)
+
+        @keypair = keypair
+        @kid = kid || generate_kid(@keypair)
+      end
+
+      def export
+        crv, x_octets, y_octets = keypair_components(keypair)
+        {
+          kty: KTY,
+          crv: crv,
+          x: encode_octets(x_octets),
+          y: encode_octets(y_octets),
+          kid: kid
+        }
+      end
+
+      private
+
+      def generate_kid(ec_keypair)
+        _crv, x_octets, y_octets = keypair_components(ec_keypair)
+        sequence = OpenSSL::ASN1::Sequence([OpenSSL::ASN1::Integer.new(OpenSSL::BN.new(x_octets, BINARY)),
+                                            OpenSSL::ASN1::Integer.new(OpenSSL::BN.new(y_octets, BINARY))])
+        OpenSSL::Digest::SHA256.hexdigest(sequence.to_der)
+      end
+
+      def keypair_components(ec_keypair)
+        encoded_point = ec_keypair.public_key.to_bn.to_s(BINARY)
+        case ec_keypair.group.curve_name
+        when 'prime256v1'
+          crv = 'P-256'
+          x_octets, y_octets = encoded_point.unpack('xa32a32')
+        when 'secp384r1'
+          crv = 'P-384'
+          x_octets, y_octets = encoded_point.unpack('xa48a48')
+        when 'secp521r1'
+          crv = 'P-521'
+          x_octets, y_octets = encoded_point.unpack('xa66a66')
+        else
+          raise "Unsupported curve '#{ec_keypair.group.curve_name}'"
```

Comment:
Missed this the first time around. The `JWT::JWKError` could be used here also

Reply:
My bad.  Should be fixed now.